cg/apps/tb/models.py

```diff
@@ -1,49 +1,41 @@
-import datetime as dt
-from pathlib import Path
 from typing import Optional
 
-from dateutil.parser import parse as parse_datestr
-from pydantic.v1 import BaseModel, validator
+from pydantic import AfterValidator, BaseModel, ConfigDict, FieldValidationInfo, field_validator
+from typing_extensions import Annotated
+
+from cg.apps.tb.validators import parse_str_to_datetime, parse_str_to_path
 
 
 class TrailblazerAnalysis(BaseModel):
     id: int
     family: str
-    case_id = Optional[str]
-
-    version: Optional[str]
-    logged_at: Optional[str]
-    started_at: Optional[str]
-    completed_at: Optional[str]
-    status: Optional[str]
-    priority: Optional[str]
-    out_dir: Optional[str]
-    config_path: Optional[str]
-    comment: Optional[str]
-    is_deleted: Optional[bool]
-    is_visible: Optional[bool]
-    type: Optional[str]
-    user_id: Optional[int]
+    case_id: Optional[str] = None
+
+    @field_validator("case_id")
+    @classmethod
+    def inherit_family_value(cls, value: str, info: FieldValidationInfo) -> str:
+        return info.data.get("family")
+
+    version: Optional[str] = None
+    logged_at: Annotated[Optional[str], AfterValidator(parse_str_to_datetime)]
+    started_at: Annotated[Optional[str], AfterValidator(parse_str_to_datetime)]
+    completed_at: Annotated[Optional[str], AfterValidator(parse_str_to_datetime)]
```

Comment:
I feel like it would make more sense to have both started_at and completed_at type hinted as `Optional[datetime.datetime]` and use a BeforeValidator instead. But given that this was not how it was done before, perhaps that should be avoided in this PR. Won't they always be returned as datetime.datetime when accessed though?

Reply:
Yeah, it is true that they will be returned as datetimes when accessed. I think having a different type hint is confusing and maybe this is something that should be refactored